lib/core/src/client/preview/StoryRenderer.tsx

```diff
@@ -230,12 +208,28 @@ export class StoryRenderer {
     }
   }
 
-  applyLayout(layout: Layout) {
-    const styles = layouts[layout] || layouts.padded;
+  applyLayout(layout: Layout = 'padded') {
+    if (layout === 'none') {
+      document.body.classList.remove(this.previousLayoutClass);
+      this.previousLayoutClass = null;
+      return;
+    }
+
+    this.checkIfLayoutExists(layout);
+
+    const layoutClass = layoutClassMap[layout];
+
+    document.body.classList.remove(this.previousLayoutClass);
+    document.body.classList.add(layoutClass);
+    this.previousLayoutClass = layoutClass;
+  }
 
-    if (styles !== this.previousStyles) {
-      Object.assign(document.body.style, styles || {});
-      this.previousStyles = styles;
+  checkIfLayoutExists(layout: string) {
+    if (!Object.keys(layoutClassMap).includes(layout)) {
```

Comment:
Any reason why not to use `if (!layoutClassMap[layout])` ?

Reply:
Good thing you added this warning by the way 👍

Reply:
should make no difference to TS

Reply:
Good catch @ghengeveld .. @guilhermewaess @yannbf ?

Reply:
Yes, I agree it's better to do !layoutClassMap, will change it tonight :)

Reply:
@ghengeveld you were right, there is a TS tricky thing here, as the layoutClassMap is not a 
`{ [key: string]: string }` 

but 

```
const layoutClassMap = {
  centered: 'sb-main-centered',
  fullscreen: 'sb-main-fullscreen',
  padded: 'sb-main-padded',
}
```

When I try to use `!layoutClassMap[layout]` it complains because the string is not a type of `'centered' | 'fullscreen' | 'padded'` 

```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ readonly centered: "sb-main-centered"; readonly fullscreen: "sb-main-fullscreen"; readonly padded: "sb-main-padded"; }'.
  No index signature with a parameter of type 'string' was found on type '{ readonly centered: "sb-main-centered"; readonly fullscreen: "sb-main-fullscreen"; readonly padded: "sb-main-padded"; }'.ts(7053)
```

So I have three options to fix it:

1st: cast the layout before use it:

```
checkIfLayoutExists(layout: Layout) {
    const layoutKey = layout as keyof typeof layoutClassMap;
    if (!layoutClassMap[layoutKey]) {
```

2nd: leave as it is now once `Object.keys().includes()` looks easier to read than the second option.
3rd: rethink about how to type this.

What do you guys think?

Reply:
Done :)